scripts/nix.conf

```diff
@@ -0,0 +1,4 @@
+# Default Nix configuration file.
+binary-caches = https://cache.nixos.org/
+signed-binary-caches = *
+binary-cache-public-keys = cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
```

Comment:
It's fine to install an empty configuration file, but it must not duplicate Nix's built-in defaults. So all the settings here should be removed.

Also, `binary-caches` has been renamed to `substituters`.

Reply:
Ah, fair enough -- let's remove it altogether then.

Reply:
On the second thought I've left it with an advice to read nix.conf manpage.

Reply:
While it may seem insane to do so from the perspective of an individual user, it's not uncommon for many enterprise environments to prefer mirroring and building everything from source in-house for security / auditing / reproducibility-without-external-internet, among other reasons.  Of course, it's certainly feasible for them to maintain a patched fork of nix, but nevertheless it'd be nice if the cache options were entirely in the configuration file instead of hard-coded in C++.  There also [a few users](https://github.com/NixOS/nixpkgs/issues/20542) who get confused when a `nix.conf` with 0 keys and 0 binary caches still substitutes from cache.nixos.org.

Just throwing this out there as food for thought; I'd imagine it's outside the scope of this particular PR.

Reply:
This is out of scope yeah, we can include `nix.conf.example` with commented options explaining what each does and what the defaults are.

Reply:
This file still needs to be removed.

Reply:
Fixed.